File: src/main/scala/com/kapilkoju/autopos/party/repo/CustomerRepo.scala
package com.kapilkoju.autopos.party.repo

import com.kapilkoju.autopos.kernel.service.AuditableBaseRepository
import com.kapilkoju.autopos.party.domain.Customer
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.EntityGraph.EntityGraphType

trait CustomerRepo extends AuditableBaseRepository[Customer] {

  @EntityGraph(value = Customer.Graph.detail, `type` = EntityGraphType.LOAD)
  def getById(id: Long): Customer

}

File: kotlang/src/main/kotlin/com/kapilkoju/autopos/party/service/CustomerRepo.kt
package com.kapilkoju.autopos.party.service

import com.kapilkoju.autopos.kernel.service.AuditableRepository
import com.kapilkoju.autopos.party.domain.Customer
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.EntityGraph.EntityGraphType

interface CustomerRepo : AuditableRepository<Customer> {

  @EntityGraph(value = Customer.Graph.detail, type = EntityGraphType.LOAD)
  fun getById(id: Long): Customer?

}
